### DISCUSSION TITLE
Dialog initial focus not working if dialog have two or more focusable elements

### DISCUSSION THREAD

**Discussion by ellemedit** [ariakit/ariakit#2700] [Questions] [answered]
[reproduction](https://stackblitz.com/edit/5ebtok?file=dialog%2Fstyle.css,dialog%2Findex.tsx&theme=light)

the `initialFocus` prop of `Dialog` has comments:

> Specifies the element that will receive focus when the dialog is first opened. It can be an HTMLElement or a React.RefObject with an HTMLElement. However, if autoFocusOnShow is set to false, this prop will have no effect. If left unset, the dialog will attempt to determine the initial focus element in the following order:
>
> 1. An element with an autoFocus prop.
> 1. The first tabbable element inside the dialog.
> 1. The first focusable element inside the dialog.
> 1. The dialog element itself.

it looks like `input` element should have initial focus but it doesn't.

could we provide alternative? such as...

```tsx
<Dialog>
  <Button />
  <DialogInitialFocusTrap>
    <input ... />
  </DialogInitialFocusTrap>
</Dialog>
```

**Comment by diegohaz** [comment, marked as answer]
The docs should clarify this, but [`autoFocus`](https://ariakit.org/reference/focusable#autofocus) will only work if used in an Ariakit component.

This works:

```jsx
<Dialog>
  <Focusable autoFocus render={<input />} />
</Dialog>
```
